src/utils/symlink.ts

```diff
@@ -39,29 +45,80 @@ export function createSymlinks(
   }
   const parsedOldVersion =
     (oldVersion ? parseVersion(oldVersion) : undefined) || undefined;
+
   const baseVersionName = path.basename(versionFilePath);
   const packageDir = path.dirname(versionFilePath);
 
-  // link latest, but only if the new version is "newer"
   if (
-    parsedOldVersion &&
-    !versionGreaterOrEqualThan(parsedNewVersion, parsedOldVersion)
+    !parsedOldVersion ||
+    versionGreaterOrEqualThan(parsedNewVersion, parsedOldVersion)
   ) {
-    logger.warn(
-      `Not updating the latest version file: current version is "${oldVersion}", new version is "${newVersion}"`
-    );
-  } else {
     logger.debug(
-      `Changing symlink for "latest.json" from version "${oldVersion}" to "${newVersion}"`
+      `${parsedOldVersion ? 'Updating' : 'Adding'} symlink for "latest.json" ${
+        parsedOldVersion ? `from ${oldVersion} ` : ''
+      }to "${newVersion}"`
     );
     forceSymlink(baseVersionName, path.join(packageDir, 'latest.json'));
   }
 
-  // link major
-  const majorVersionLink = `${parsedNewVersion.major}.json`;
-  forceSymlink(baseVersionName, path.join(packageDir, majorVersionLink));
+  // Read possibly existing symlinks for major and minor versions of the new version
+  const newVersionMajorSymlinkedVersion = getExistingSymlinkedVersion(
```

Comment:
this variable name seems overly verbose and also incorrect -- maybe `existingMajorSymlink` instead ? (same for the other one right below this)